src/main/java/com/chargerrobotics/subsystems/LimelightSubsystem.java

```diff
@@ -70,21 +93,145 @@ public double getV() {
 		return v;
 	}
 
-	// distance in inches
-	public double distance() {
-		if (v == 0) {
-			return -1.0;
+	public double getS() {
+		return s;
+	}
+
+	public int pl;
+	public double distM;
+	public double skewangle;
+	public double xcoord;
+	public double ycoord;
+	public double realangle;
+	public double compassdeg;
+
+	
+	public int findPL() {
+		for (int b = 0; b <= 10; b++) {
+			pip.setNumber(b);
+			pl = b;
+			if (v == 1)
+				break;
+			// cycles through pipelines looking for a target
+		}
+
+		return pl;
+
+	}
+	// finds distance to markers
+	public Double distanceM() {
+		findPL();
+		if (pl == 0 || pl == 1)
+			return null;
+		distM = Constants.cameraHeight * (Math.tan(Math.toRadians(Math.abs(y - Constants.cameraAngle))));
+		return distM;
+	}
+
+	// returns x coordinates in inches
+	public Double locX() {
+		double skew;
+
+		findPL();
+		if (pl == 2 || pl == 4) {
+			skew = s * -1.0;
+			realangle = skew + x;
+		} else if (pl == 3 || pl == 5) {
+			skew = s + 90.0;
+			realangle = skew - x;
+		} else {
+			return null;
+		}
+		switch (pl) {
+			case 0:
+				return null;
+			case 1:
+				return null;
+			case 2:
+				xcoord = onex;
+			case 3:
+				xcoord = twox;
+			case 4:
+				xcoord = threex;
+			case 5:
+				xcoord = fourx;
+		}
+		Double X = distanceM() * Math.cos(Math.toRadians(realangle)) + xcoord;
+		return X;
+	}
+
+	// returns y coordinantes in inches
+	public Double locY() {
+		double skew;
+
+		findPL();
+		if (pl == 2 || pl == 4) {
+			skew = s * -1.0;
+			realangle = skew + x;
+		} else if (pl == 3 || pl == 5) {
+			skew = s + 90.0;
+			realangle = skew - x;
+		} else {
+			return null;
+		}
+		switch (pl) {
+			case 0:
+				return null;
+			case 1:
+				return null;
+			case 2:
+				ycoord = oney;
+			case 3:
+				ycoord = twoy;
+			case 4:
+				ycoord = threey;
+			case 5:
+				ycoord = foury;
+		}
+		Double Y = distanceM() * Math.sin(Math.toRadians(realangle)) + ycoord;
+		return Y;
+	}
+
+	// finds absolute rotation of robot on field,
+	public Double degrees() {
+
+		findPL();
+		switch (pl) {
+			case 0:
+				return null;
+			case 1:
+				return null;
+			case 2:
+				compassdeg = 270.00 - Math.abs(s); // marker 1
```

Comment:
Don't forget to put a `break;` statement at the end of your case statements so that it doesn't fall down to the next case.